pkg/registry/core/pod/strategy.go

```diff
@@ -170,7 +170,7 @@ func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, bool, error) {
 	if !ok {
 		return nil, nil, false, fmt.Errorf("not a pod")
 	}
-	return labels.Set(pod.ObjectMeta.Labels), PodToSelectableFields(pod), pod.Initializers != nil, nil
+	return labels.Set(pod.ObjectMeta.Labels), apiv1.PodToSelectableFields(pod), pod.Initializers != nil, nil
```

Comment:
@deads2k can you weigh in on this approach... I know you had thoughts on improving the field selector conversion. Do we really want our registries pinned to specific API versions?

Reply:
I really don't...  field selectors are part of the API

Reply:
@liggitt @smarterclayton Due to current inconsistency field exposure between [`addConversionFuncs in pkg/api/v1`](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/conversion.go#L132-L223) and `pkg/registry/core/xxx/strategy.go`, we should enforce the same rules to avoid issues like #50234 and #53362. That's why I move those related common functions to `pkg/api/v1` instead of keeping in `pkg/registry/core/xxx`, which will result in import cycle if we use them in `pkg/api/v1`. And maybe we can replace them in other places. WDYT?

Reply:
We do not want our selectors pinned to v1.

There needs to be conversion of selectors in the high level api (names) that translates into errors thrown by storage.  No code outside of storage should be rejecting field selectors (it should not be done in validation, nor in the API).  The mapping is acceptable to do in conversion functions in various versions, but the extra complexity moving this out of storage isn't justified.